inst/include/Rcpp/platform/compiler.h

```diff
@@ -24,173 +22,35 @@
 
 // NB: A vast list valid identifiers is at these wiki pages:
 //     http://sourceforge.net/p/predef/wiki/Home/
-
-#undef GOOD_COMPILER_FOR_RCPP
-#ifdef __GNUC__
-#define GOOD_COMPILER_FOR_RCPP
-#endif
-#ifdef __SUNPRO_CC
-#define GOOD_COMPILER_FOR_RCPP
-#endif
-#ifdef __clang__
-#define GOOD_COMPILER_FOR_RCPP
-#endif
-#ifdef __INTEL_COMPILER
+#if defined(__GNUC__) || defined(__SUNPRO_CC) || defined(__clang__) || defined(__INTEL_COMPILER)
 #define GOOD_COMPILER_FOR_RCPP
+#else
+#error "This compiler is not supported"
```

Comment:
And just error? GOOD_COMPILER_FOR_RCPP is not used anywhere else and packages should not rely on this (not documented, no RCPP_ prefix).

Reply:
And I just checked and nobody uses it on CRAN.

Reply:
Compare to what it in the mainline or CRAN version. I just wanted a TRUE body as the alternative is to write `!defined(ONE) && !defined(ANOTHER)` etc and I did not like that at the time.   Can revisit.

Reply:
Addressed in next commit.